test/writer.js

```diff
@@ -41,39 +41,51 @@ describe('#unit Check database restore writer', function() {
     nock.cleanAll();
   });
 
-  it('should complete successfully', function(done) {
+  it('should complete successfully', async function() {
     nock(dbUrl)
       .post('/_bulk_docs')
       .reply(200, []); // success
 
-    testLinestream()
-      .pipe(writer(db, 500, 1, noopEmitter))
-      .on('error', function(err) {
-        done(err);
-      })
-      .on('finished', function(data) {
-        assert.strictEqual(data.total, 15);
-        assert.ok(nock.isDone());
-        done();
-      });
+    return new Promise((resolve, reject) => {
```

Comment:
As you mentioned to me I think we can use pipeline here, it would look something like:
```
    const w = writer(db, 500, 1, noopEmitter);
    return Promise.all([pipeline(fs.createReadStream('./test/fixtures/animaldb_expected.json'), liner(), w),
      once(w, 'finished').then((data) => {
        assert.strictEqual(data.total, 15);
        assert.ok(nock.isDone());
      })]);
```